### PR TITLE
ignore certain errors

### PR DESCRIPTION
**Pull request by cyx**
Don't report certain errors

### CODE REVIEW

File: rollrus.go [hunk 1]
```diff
@@ -26,20 +26,47 @@ var wellKnownErrorFields = []string{
...
 	roll.Client
-	triggers []log.Level
+	triggers      []log.Level
+	ignoredErrors map[error]bool
+}
+
+// OptionFunc is any option you can pass to NewHook.
+type OptionFunc func(*Hook)
+
+// WithLevels is an option to allow you to pass in levels.
```
Comment (freeformz) [inline rollrus.go]: And what happens to the passed in levels?

File: rollrus.go [hunk 2]
```diff
@@ -26,20 +26,47 @@ var wellKnownErrorFields = []string{
 // May be used as a rollbar client itself
 type Hook struct {
 	roll.Client
-	triggers []log.Level
+	triggers      []log.Level
+	ignoredErrors map[error]bool
```
Comment (freeformz) [inline rollrus.go]: `map[error]struct{}` ?

File: rollrus_test.go [hunk 3]
```diff
@@ -154,3 +155,12 @@ func TestTriggerLevels(t *testing.T) {
 		t.Fatal("Expected Levels() to return newLevels")
 	}
 }
+
+func TestWithIgnoredErrors(t *testing.T) {
+	hook := NewHook("foobar", "testing", WithIgnoredErrors(io.EOF))
+
+	want := map[error]bool{io.EOF: true}
+	if !reflect.DeepEqual(hook.ignoredErrors, want) {
```
Comment (freeformz) [inline rollrus_test.go]: Is this test useful? It's basically testing that you set some internal variable vs. testing to see if logging with the hook skips io.EOF errors. Also a negative test would also be ++

File: rollrus.go (hunk 2 above)
Comment (cyx) [inline rollrus.go]: `map[error]struct{}`  is meh because the use will be `if _, ok` instead of `if ignoredErrors[err]`. Yay / nay?

**Comment by cyx** [comment]
@freeformz addressed comments ^

File: rollrus.go (hunk 2 above)
Comment (freeformz) [inline rollrus.go]: What does true and false mean?

If you want to check for existence do it like so:

```go
if _, ok := map[thing]; ok {
    // ....
}
```

**Pull request merged**